iquip/apps/builder.py

```diff
@@ -309,11 +312,83 @@ def __init__(
 
         Args:
             name: See the attributes section.
-            argInfo: See the attributes section.
+            argInfo: Each key and its value are:
+              default: The dictionary that describes arguments of a scannable object.
+              unit: The unit of the number value.
+              scale: The scale factor that is multiplied to the number value.
+              global_step: The step between values changed by the up and down button.
+              global_min: The minimum value. (default=float("-inf"))
+              global_max: The maximum value. (default=float("inf"))
+                If min > max, then they are swapped.
+              ndecimals: The number of displayed decimals.
         """
         super().__init__(parent=parent)
         self.name = name
-        self.argInfo = argInfo
+        self.state = self.get_state(argInfo)
+        self.stack = QStackedWidget(self)
+        self.layout = QGridLayout(self)
+        self.layout.addWidget(QLabel(name, self), 0, 0)
+        self.layout.addWidget(self.stack, 1, 1)
+
+    def get_state(self, argInfo):
+        """Gets a state dictionary that describes key parameters of a scannable object.
+
+        Creates a default state dictionary and updates it using the argInfo.
+
+        Args:
+            argInfo: A dictionary that contains argument information of each scannable type.
+        """
+        scale = argInfo["scale"]
+        state = {
+            "selected": "NoScan",
+            "NoScan": {"value": 0.0, "repetitions": 1},
+            "RangeScan": {"start": 0.0, "stop": 100.0*scale, "npoints": 10,
+                          "randomize": False, "seed": None},
+            "CenterScan": {"center": 0.*scale, "span": 100.*scale,
```

Comment:
Please replace "span" with "step" same as #111.

Reply:
Do you want me to change variable name?
What do you want to say with this #111 reference?

Reply:
This PR has too many content.
Can you explain me in more detail?

Reply:
Oh, I'm really sorry that I confused you.

I intended to say snu-quiqcl/artiq-proxy#111.
I want just to replace the positions of "span" and "step" as follows:
```python
{"center": ..., "step": ..., "span": ..., ...}
```